testing/python/layout/test_tilelang_cute.py

```
@@ -477,6 +587,167 @@ def test_logical_divide_dynamic_layout_stride():
+    # A ScaledBasis-strided layout maps to (datapath, column) coordinates.
+    # restrict returns the region origin's coordinates plus the sliced
+    # sublayout, and the affine identity holds componentwise.
+    restride = cute.make_layout((128, 16384), stride=(cute.E(0), cute.E(1)))
+    frag = cute.composition(restride, cute.make_layout(((16, 4), 32), stride=((1, 32), 128)))
+    assert frag((16, 0)) == (32, 0)
+    offset, tile = cute.restrict(frag, [tvm.ir.Range.from_min_extent(0, 64), tvm.ir.Range.from_min_extent(8, 24)])
+    assert offset == (0, 8)
+    assert tile((0, 3)) == (0, 3)
+    for i in range(0, 64, 7):
+        for j in range(0, 24, 5):
+            assert frag((i, 8 + j)) == tuple(base + step for base, step in zip(offset, tile((i, j))))
```

Ruff (0.16.0)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)